scripts/core/editor3/components/Editor3Component.tsx

```diff
@@ -603,39 +658,71 @@ export class Editor3Component extends React.Component<IPropsEditor3Component, IS
             }
         };
 
+        const inputStyle: React.CSSProperties = (() => {
+            const style: React.CSSProperties = {};
+
+            if (this.props.uiTheme != null) {
+                style.fontSize = this.props.uiTheme.fontSize;
+                style.color = this.props.uiTheme.textColor;
+                style.fontFamily = this.props.uiTheme.fontFamily;
+            }
+
+            if (this.state.expanded) {
+                return {...style, height: 'auto'};
+            } else if (this.props.expandable?.enabled) {
+                return {...style, overflowY: 'auto', maxHeight: this.collapsedHeight};
+            }
+
+            return {...style};
+        })();
+
         return (
-            <div
-                className={cx}
-                ref={(div) => this.div = div}
-                onDragStart={() => {
-                    if (this.state.draggingInProgress !== true) {
-                        setTimeout(() => {
+            <Spacer v gap="0" justifyContent="center" alignItems="center" noWrap>
+                <div
+                    className={cx}
+                    ref={(div) => {
+                        this.div = div;
+
+                        // if element gets resized from somewhere we should
+                        // check if the expand button should still show
+                        if (div) {
+                            const resizeObserver = new ResizeObserver(() => {
```

Comment:
ref function may run multiple times - ensure you do not end up with more than one observer.

I was about to suggest using lodash.once, but that would be incorrect too, because on every run, the element reference might be a different one and if you cache the first one, you would end up with a stale reference.

You'll need to disconnect old observer here and connect a new one.